src/renderer/components/Button/Button.tsx

```diff
@@ -1,12 +1,44 @@
 import * as React from "react";
 
-interface IProps {
-    type: string;
-    children: any;
+export interface IButtonProps {
+    disabled?: boolean;
+    onClick?: () => {};
 }
+export interface IBaseButtonProps extends IButtonProps{
+    buttonType: string;
+}
+const TYPES = {
+    PRIMITIVE: "primitive",
+    PRIMARY: "primary",
+    SECONDARY: "secondary",
+    INVERTED: "inverted",
+    DESTRUCTIVE: "destructive",
+};
+
+const BaseButton: React.FunctionComponent<IBaseButtonProps> = ({
+    children,
+    disabled,
+    onClick,
+    buttonType,
+}) => (<button 
+    className={`button btn-${buttonType}`} 
+    disabled={disabled} 
+    onClick={onClick}>
+    {children}
+    </button>);
 
-export const Button: React.FunctionComponent<IProps> = ({type, children, ...props}) => (
-    <button className={`button btn-${type}`} {...props}>
-        {children}
-    </button>
-);
+export const ButtonPrimitive: React.FunctionComponent<React.PropsWithChildren<IButtonProps>> = ({children, disabled, onClick}) => {
+    return(<BaseButton disabled={disabled} onClick={onClick} buttonType={TYPES.PRIMITIVE}>{children}</BaseButton>);
```

Comment:
To reduce code duplication you can create method like:

renderSpecificButton(type: string, props: IButtonProps) {
 return(<BaseButton disabled={props.disabled} onClick={props.onClick} buttonType={type}>{children}</BaseButton>);  
}